service/service.go

```diff
@@ -120,12 +65,20 @@ func New(
 		blockType:                 blockType,
 		parallelSubRequests:       parallelSubRequests,
 		blockRangeSizeSubRequests: blockRangeSizeSubRequests,
+		requestValidator:          NewRequestValidator(blockType),
 		tracer:                    tracer,
 	}
+
 	zlog.Info("creating gprc client factory", zap.Reflect("config", substreamsClientConfig))
 	clientFactory := client.NewFactory(substreamsClientConfig)
 
-	s.workerPool, err = orchestrator.NewWorkerPool(parallelSubRequests, clientFactory)
+	workerFactoryGenerator := func() orchestrator.WorkerFactory {
+		return func(tracer ttrace.Tracer) orchestrator.Worker {
+			return orchestrator.NewRemoteWorker(clientFactory, tracer)
+		}
+	}
```

Comment:
that seems a bit convoluted to pass down an instance of a tracer, isn't it? Couldn't that be passed as a variable somewhere in the stack?
The tracer instance is pretttty abstract and high up.. could be known any top sort of top-level objects.

Perhaps we're in need of an `Orchestrator` object? Something that would tie up all those loose ends in `backprocess.go` huh? :)